### PR TITLE
feat: export and import whole device configuration (#1924)

### LINKED ISSUE #1924: 🚀 [Feature Request]: export whole configuration
### OS

iOS

### Description

Android app lets users export full device config (this is also possible from CLI). Please add that feature to iOS/macOS APP

### Participation

- [ ] I am willing to pay to sponsor this feature.
- [ ] I am willing to submit a pull request for this feature.

### Additional comments

_No response_

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bruschill**
## What changed?

Adds **Export** and **Import** of a node's whole configuration to **Settings → Tools**, using the binary `DeviceProfile` (`.cfg`) format shared with the Android app and the `meshtastic` CLI.

> **Supersedes #1940.** That PR was export-only; this one carries the same export work **plus** the inverse import feature so the two ship together. Closes #1924.

### Export

- `DeviceProfileExport.swift` — entity → protobuf converters for all 8 device configs and 15 module configs (the inverse of the proto → entity mapping in `UpdateSwiftData.swift`), plus `NodeInfoEntity.exportDeviceProfile()` which assembles a `DeviceProfile`: long/short name, `LocalConfig`, `LocalModuleConfig`, the channel-set URL, ringtone, canned messages, and the fixed position when one is set.
- `DeviceProfileDocument.swift` — a `FileDocument` wrapping the serialized profile bytes for `.fileExporter`, with an Android-style default filename and a shared `.meshtasticDeviceProfile` UTType.
- `Tools.swift` — the export button + `fileExporter`, gated on a connected node, behind a confirmation dialog that warns about the sensitive material in the file.
- `UpdateSwiftData.swift` — persists MQTT `mapReportingShouldReportLocation` in the config-update branch (it was only set on insert) so the exported value isn't a stale default.

### Import

- `DeviceProfileImport.swift` — pure, device-free core. `parseDeviceProfile` defensively decodes untrusted bytes (size cap **before** decode, capped read at the call site). `DeviceProfileImportPlan(profile:currentUser:)` is the `has*`-gated **inverse of `exportDeviceProfile()`** — the single source of coverage truth so export/import can't silently drift. It emits an ordered `[ImportItem]` plan grouped into 8 coarse toggle sections, each item carrying `isSensitive`/`causesReboot` flags.
- `DeviceProfileImporter.swift` — a `ProfileApplyGateway` seam plus the apply engine (sequential, `isConnected` pre-checks, abort-on-first-failure). The production gateway forwards each item to the existing `save*Config` / `saveUser` / `saveChannelSet` / `saveRtttlConfig` / `saveCannedMessageModuleMessages` / `setFixedPosition` calls, with `fromUser == toUser ==` the connected node (self-admin, no session passkey).
- `ImportDeviceProfileView.swift` — review sheet: per-section toggles, live sensitive-material + reboot banners, a confirmation gate, per-item progress, and an applied/failed/skipped result summary.
- `Tools.swift` — the Import section + `.fileImporter([.meshtasticDeviceProfile])` with a capped read, parse, and the review sheet.
- `AccessoryManager+ToRadio.swift` — new `setFixedPosition(_:fromUser:toUser:)` overload to restore explicit coordinates from a profile (the existing overload only uses the phone's live GPS fix).

Import design decisions worth a look:

- **Apply order is fixed and reboot-safe.** The plan emits items in a strict order (owner → radio/device → modules → personalization → fixed position → network → bluetooth → security → **Channels & LoRa last**). The UI filters by section but never reorders, so the reboot-prone LoRa/channel step is always last and can't drop later sends. Network/Bluetooth are placed late because they can drop the transport the import runs over.
- **Security & Identity defaults OFF.** Importing a foreign private/admin key rewrites this node's cryptographic identity and can break existing DMs / lock local admin, so it's a deliberate opt-in. Every other present section defaults ON, preserving a one-tap "restore my device" flow.
- **LoRa/channel dedupe.** Export writes the *same* `LoRaConfig` into both `config.lora` and the channel URL's `ChannelSet`. The plan emits exactly one terminal item — the channel URL when it's a valid replace-mode set with a LoRa config, otherwise the standalone LoRa config (add-mode or LoRa-less URLs fall back so the region is still restored).
- **Owner merge preserves identity.** The owner step starts from the live `node.user.toProto()` and overrides only long/short name, so `id`/`hwModel`/`publicKey`/`role` are never wiped by `setOwner`.
- **Partial imports are safe and escapable.** An adversarial safety pass confirmed the device is left coherent and recoverable in every partial-failure case (mid-import disconnect, reboot, or a rejected send) — no bricking, no sends into a dead link, no crash. On top of that: the apply is now cancellable (a Cancel button on the applying screen, per-item `Task.isCancelled`/`CancellationError` handling, and a liveness backstop so a single stalled send can never trap the sheet); the result reports what was **sent** (admin sends are fire-and-forget, not device-acked), always nudges the user to reconnect and verify, and warns when a destructive channel replace may have partially applied. `ImportItemKind.displayName` gives human-readable labels in progress/results instead of raw case names.

### Applying inside a firmware edit transaction

Hardware testing showed the original per-item apply was unsafe in practice, so the apply is now bracketed by the firmware's edit transaction. This is the most significant change since the last review.

- `DeviceProfileImporter.swift` — the run is `begin_edit_settings` → items → `commit_edit_settings`. With a transaction open the firmware defers every flash save, reboot, and Bluetooth teardown to the commit, so a full profile collapses from N saves + N reboots into **one save and one reboot**. Verified on hardware: 19 config writes produced 20 `Delay save of changes to disk until the open transaction is committed` lines and exactly one `Save changes to disk` + one `Reboot`.
- **The commit is sent unconditionally**, even when sending stopped early. The firmware exposes no abort or rollback, and an uncommitted transaction leaves the node in deferred-save mode where nothing *any* client writes afterwards is persisted, until it is committed or rebooted. Committing a partial profile matches the pre-transaction behaviour of aborting on first failure, and is strictly better than stranding the node. `commitUnconfirmed` distinguishes "the link was already down so the node may still hold the transaction open" from the normal case where the commit's own reboot eats the ack.
- **`begin_edit_settings` is sent twice.** A dropped begin is undetectable from the client and silently downgrades the whole run to untransacted — every write then saves to flash, tears down Bluetooth, and schedules its own reboot. That was observed on hardware. A duplicate begin is accepted by the firmware when a transaction is already open (confirmed on 2.7.27 and 2.8.0), so the second copy costs one packet and removes a single point of failure.
- **Sends are paced** (150 ms). The radio drops writes it cannot process in time and still acks them, so an unpaced burst loses config silently.
- **MQTT and Serial are applied after the commit**, outside the transaction, because the commit calls `disableBluetooth()` and the BLE link is gone before they could be sent. Over BLE they therefore usually cannot run at all and are reported as needing a reconnect and a second pass — not as failures, since everything else is already committed.

### Post-import verification

- `DeviceProfileVerifier.swift` + the result screen compare what the radio now reports against what was sent, per section. Admin sends are fire-and-forget, so without this the UI could only report what it *sent*.
- Verification is gated on the radio having sent a fresh configuration since the import finished (`lastConfigRefresh >= importFinishedAt`), checked both in the view and again inside the verifier. Running it earlier compares against the pre-import cache and reports every section as lost, which is indistinguishable from total failure. This is not theoretical: the firmware only sends config during the initial `want_config` handshake, so a connected client's cache is stale the moment anything changes it externally.

### Fixes found while testing this

These are bugs the hardware testing surfaced. Three are not import-specific.

- **`3adedc8f`** — Settings → Tools, and with it this entire feature, was hidden on devices without NFC. The feature was unreachable on affected hardware.
- **BLE transport — now [#2215](https://github.com/meshtastic/Meshtastic-Apple/pull/2215), split out of this PR.** A single write the radio could not allocate for (`CBATTError.insufficientResources`) was escalated to a fatal session teardown with no reconnect. Found here, but it affects any large admin message — renaming a node, most config writes — so it is reviewed separately. **Not part of this PR's diff.**
- **`6544c46a`** — the in-app log viewer filtered on a hard-coded subsystem string while `Logger` derives it from `Bundle.main.bundleIdentifier`, so any contributor building under their own bundle id saw an empty log with no indication why.
- **`709fd9d0`** — export wrote files with **no extension**, so an exported profile could not be re-imported. `UTType(filenameExtension:)` returns nil for an extension no installed app has declared, and the `?? .data` fallback silently substituted a type with no extension; separately, `fileExporter` does not append `preferredFilenameExtension` to `defaultFilename` at all. Fixed by declaring `com.meshtastic.device-profile` in `Info.plist` and putting `.cfg` in the filename, which also matches Android's naming exactly.

### Localization

- `ImportItemKind.displayName` and `ImportSection.title` were plain `String`s shown via `Text(variable)`, which SwiftUI does **not** localize — routed them through `.localized` with casing that matches existing config-screen strings, so 25 shared labels reuse their translations. The assembled confirmation/sensitive-material messages are localized via `.localized` fragments + `String(format:)`.
- Registered all new import (and previously-unlocalized export) UI strings in `Localizable.xcstrings` as source-language entries (English default = key), so translators / `scripts/translate-locale.sh` can fill the other 17 locales.

### Docs

- Adds a **Tools** section to `docs/user/settings.md` (NFC contact tag, Export, and Import) — the Tools screen was previously undocumented — and regenerates the bundled in-app HTML.

## Why did it change?

The Android app and `meshtastic` CLI can export **and** import a device's full configuration; iOS/macOS could do neither. The `.cfg` binary `DeviceProfile` format matches Android, so a profile can be backed up, restored, or cloned onto another node across the Meshtastic ecosystem. Shipping export and import together closes the loop in one change.

### Automated

- **76 tests in 4 suites**, all passing: `DeviceProfileExportTests`, `DeviceProfileExportFilenameTests`, `DeviceProfileImportTests`, `DeviceProfileVerifierTests`.
- Full app suite run and compared against the merge base: **2,620 tests in 460 suites** on this branch vs **2,544 in 456** at the merge base (+76 / +4, matching the suites added here). Two tests fail — `allCases_count()` and `validNumCreatesUserWithDefaults()` — and **both fail identically at the merge base**, so this branch introduces no new failures. Diffing the two failure sets returns empty.
- Import-engine coverage includes the transaction bracket, the paced/duplicated begin, commit-on-failure, commit-into-a-dead-link (`commitUnconfirmed`), the post-commit MQTT/Serial phase, cancellation, and both stale-readback guards on verification.

### On hardware, over BLE, from a physical iPhone

Every scenario below was **driven by hand through the real app on a physical iPhone** — pairing to the radio over BLE, exporting, picking the `.cfg` in the file browser, choosing sections, tapping Apply, and (for the interruption cases) killing the phone's radio mid-run from Control Center. These are real UI flows, not synthetic admin sends.

The radio side was instrumented separately: a Meshtastic MCP server handled setup and verification (config snapshots, before/after `config_diff`, enabling `debug_log_api`, reboots), and a serial capture recorded the firmware's own log throughout. So each result is corroborated by what the firmware logged, not by what the app claimed on screen — which matters, because one of the bugs found here was the app reporting success for a write that never reached the radio.

Two Heltec V4 (ESP32-S3) units, one on each firmware.

| # | Scenario | Firmware | Result |
| --- | --- | --- | --- |
| 1 | Clean import | 2.7.27 | 19 sets → 20 deferred saves → **1** flash save → **1** reboot |
| 2 | Config round-trip | 2.7.27 | `config_diff` returns to the pre-import snapshot exactly |
| 3 | Link lost mid-import | 2.7.27 | Stranded state reproduced: begin, sets, no commit. App reported the failure and the unconfirmed-commit warning rather than claiming success |
| 4 | Recovery from stranded | 2.7.27 | Re-import commits and flushes the pending writes; a reboot instead discards them |
| 5 | Verify-before-reconnect | 2.7.27 | Blocked by both guards; also physically impossible over BLE, since commit disables Bluetooth before any refresh can arrive |
| 6a | Same import on 2.8 | **2.8.0** | Transaction semantics **identical** to 2.7.27; the 2.8-only `traffic_management` section gated in and applied (20 settings vs 19) |
| 6b | Untransacted contrast | 2.8.0 | An ordinary LoRa write from the settings screen saves to flash **5 ms** later with zero deferrals — the case the transaction exists to prevent |
| 6c | Cross-version 2.8 → 2.7.27 | 2.7.27 | `traffic_management` correctly **not sent**; `.fromVersion("2.8.0")` blocked it client-side rather than letting the radio silently discard it |
| 7 | Same import on **nRF52** (Seeed T1000-E, nRF52840/SoftDevice) | 2.8.0 | 20 transacted settings → **1** flash save → **1** reboot. Transaction semantics identical to ESP32 |

Scenario 6a is the one worth noting for reviewers: **the recent firmware reboot rework does not change the transaction path.** Deferred saves, a single flash write, a single reboot, and Bluetooth teardown at commit behave the same on 2.7.27 and 2.8.0-develop.

The 2.8.0 rows carry a caveat: that radio was dropping BLE notifications throughout ([meshtastic/firmware#11245](https://github.com/meshtastic/firmware/issues/11245)). Scenario 6a passed while 694 notifications were dropped, and an earlier attempt at the same import aborted mid-run on the same radio and had to be recovered by rebooting it. Treat 2.8.0 results here as "passed over a lossy link" rather than "passed cleanly".

### Not tested

- **Three radios, two chip families**: 2× Heltec V4 (ESP32-S3, NimBLE) and 1× Seeed T1000-E (nRF52840, SoftDevice). No other hardware.
- The BLE retry found during this work now lives in [#2215](https://github.com/meshtastic/Meshtastic-Apple/pull/2215) and is not tested here.
- Scenario 3 was reproduced by killing the phone's radio, not by genuine range loss.
- Import reliability on 2.8.0 is confounded by [meshtastic/firmware#11245](https://github.com/meshtastic/firmware/issues/11245). One import on that firmware failed partway and left the node holding an open transaction; the same file succeeded after a reboot. I could not separate "2.8 is fine" from "that radio was out of BLE buffers" with the hardware available.

### Platform difference worth knowing (ESP32 vs nRF52)

The post-commit MQTT/Serial phase behaves differently per chip, and the design handles both — but reviewers should know the outcomes differ:

- **ESP32 (NimBLE).** `disableBluetooth()` calls `nimbleBluetooth->deinit()`, a hard synchronous teardown. The BLE link is gone before MQTT/Serial can be sent, so they are reported as needing a reconnect and a second pass. **One reboot total.**
- **nRF52 (SoftDevice).** `nrf52Bluetooth->shutdown()` leaves the link usable long enough for both to be delivered, so they apply in the same run and no second pass is reported. Each is outside the transaction by design, so each performs its own save and reboot — **three reboots total** (one for the transaction, one each for MQTT and Serial).

Both are correct and both are much better than untransacted (which would be ~22 saves and reboots), but the user-visible result differs: an nRF52 user sees the profile fully applied with extra reboots, an ESP32 user sees two items pending a second pass.

Two nRF52-specific firmware behaviours were also checked:

- `NRF52Bluetooth.cpp` **deduplicates consecutive identical ToRadio writes**, which looked like a threat to the duplicated `begin_edit_settings`. It is not: `Drop dup ToRadio packet` never fired and both `Begin transaction` lines were logged. (Even if it had, the safety net survives — a begin lost in transit never updates `lastToRadio`, so the retry is not seen as a duplicate.)
- Under the 22-message burst the nRF52 reports `fromRadioQ full, drop oldest!` (×19) and `tophone queue status queue is full, discard oldest` (×22). The import still completed correctly, but the radio→phone path saturates, which also means `debug_log_api` output is lossy during a burst on this hardware — worth knowing when reading nRF52 logs.

### Reproducing the key scenarios by hand

Anyone with two radios and a phone can re-run the important ones:

**Transaction collapse (scenario 1).** Connect, Settings → Tools → Import, pick a `.cfg`, select all, Apply. With `debug_log_api` on, the serial log should show `Begin transaction` twice, one `Delay save of changes to disk until the open transaction is committed` per item, then exactly one `Save changes to disk` and one `Reboot`. Seeing more than one save means the transaction did not take.

**Stranded node (scenario 3).** Start the same import, then turn on Airplane mode about a third of the way through. Expect the result screen to report the failure plus "The node did not confirm saving these settings" — **not** a success. The firmware log should show the begin and several sets with no commit. The node then looks healthy but silently discards config from every client until it is committed or rebooted.

**Recovery (scenario 4).** From that stranded state, reconnect and import the same file again. It should complete and commit normally, which is what the result screen's advice tells the user to do.

**Untransacted contrast (scenario 6b).** Change a single LoRa field from the ordinary Settings → LoRa screen. That write saves to flash immediately, with no deferral — the behaviour the transaction exists to avoid across a whole profile.

Note the interruption window is only about 8 seconds at the shipped 150 ms pacing, so scenario 3 takes a couple of tries to land by hand.

## Checklist

- [x] My code adheres to the project's coding and style guidelines.
- [x] I have conducted a self-review of my code.
- [x] I have commented my code, particularly in complex areas.
- [x] I have verified whether these changes require updates to the in-app documentation under `docs/user/` or `docs/developer/`, and updated accordingly. Added a **Tools** section to `docs/user/settings.md` (NFC, Export & Import Configuration) and regenerated the bundled HTML.
- [x] I have tested the change to ensure that it works as intended.

---

### Notes for reviewers

- **Sensitive material is exported in plaintext and written back on import.** The profile includes the private key, admin keys, channel PSKs, and Wi-Fi/MQTT passwords. Export only reads it; import *applies* it, which is strictly more dangerous — so on import, Security & Identity defaults OFF, a live banner itemizes exactly what a selection overwrites, and a confirmation dialog gates apply. This matches Android's `.cfg` behavior (the format is meant to fully restore a device). No secret is logged.
- **Export ↔ import are two hand-maintained mirrors.** `DeviceProfileImportPlan` is the inverse of `exportDeviceProfile()`, and each export converter is the inverse of an `upsert*ConfigPacket` mapping; a round-trip test locks the format, but a shared single source of truth between the directions would be the deeper fix for a future pass.
- **Partial-import safety.** The apply is now wrapped in the firmware's edit transaction (see above), which changes the guarantee from the earlier revision of this PR: it is no longer per-item non-atomic. A full run is one save and one reboot. A run that fails partway still commits what landed, deliberately — the alternative is leaving the node in deferred-save mode where it silently discards config from every client until rebooted. The import remains safe to re-run, every send is idempotent, and re-importing is also the recovery path for a node left stranded by a dropped link.

- **A stranded node is a real state and users can reach it.** If the transport dies before the commit, the node stays connected and looks healthy while discarding every config write. The result screen surfaces this, and the recovery (re-import) was verified on hardware. Worth a look at whether that warning is prominent enough.

- **Firmware issue found during this work: [meshtastic/firmware#11245](https://github.com/meshtastic/firmware/issues/11245).** On 2.8.0 / ESP32-S3, BLE notifications fail continuously with `BLE_HS_ENOMEM` (`rc=6`) at 1.27–1.45/s while a client syncs, starting moments after boot. A 2.7.27 control on the same hardware model logged **zero** occurrences over 61 minutes. Every failure is a dropped `FromRadio` notification. This is a firmware-side condition this PR cannot fix, but it means **the app is operating over a lossy link on 2.8**, which is useful context if import flakiness is reported. Caveat stated in the issue: two different physical units with different NodeDB sizes, no bisect, so suggestive of a regression rather than proof. Notably scenario 6a *succeeded* while 694 notifications were dropped — client-visible success is not evidence of a healthy link.

- **Split: [#2215](https://github.com/meshtastic/Meshtastic-Apple/pull/2215) (BLE retry on out-of-buffers).** Found while testing this feature, but it is a transport-layer change affecting every BLE write in the app, so it is now its own PR against `main` and is **not** in this diff. There is no code dependency in either direction — it touches only `BLEConnection.swift` and nothing here references it — so the two can land in any order. Landing #2215 first is preferable only because this PR's config-edit transaction raises the cost of a mid-import transport failure: with a transaction open, a link that dies before the commit leaves the node holding it, and #2215 reduces how often that happens on ESP32.

## Summary by CodeRabbit

* **New Features**
  * Added device configuration export to `.cfg` files (compatible with Android app and the Meshtastic CLI).
  * Added configuration import with a section-by-section review, sensitive-data warnings, progress feedback, and safe apply behavior that stops on the first failure (including reboot-relevant LoRa/Channels & LoRa).
  * Added Settings → Tools utilities for export/import and creating Node Contact NFC tags.
* **Bug Fixes**
  * Improved persistence for admin/security keys, MQTT location reporting, and Store & Forward server mode.
* **Documentation**
  * Updated Settings and developer deep-link docs for supported URL formats and expanded TAK/settings routes.

### CODE REVIEW

**Comment by tylerpieper** [comment]
Android will import a partial config - if a cfg doesn't have the longname populated for example, it will leave it as is on the node. 

I haven't read your code yet but I want to ask that you also support a partial config import. I see a future where events and stuff can just pass an official config file that sets everything except names and keys and stuff instead of needing to do event-specific firmware and stuff. 

My next PR will be for CLI to export a partial config to create these

**Comment by bruschill** [comment]
@coderabbitai review

**Comment by bruschill** [comment]
## Field-level export parity: 16 active proto fields are unmapped

While restoring my test radios after the hardware runs I diffed each node against a pre-import snapshot, and found that importing a profile had silently changed `power.sds_secs` from `4294967295` to `0` — **including on a node importing its own export**. A faithful round trip changed the config.

The cause is not a bug in the transaction or the apply path. `PowerConfigEntity.protoConfig` simply never assigns `sdsSecs`, so it is serialized as the proto default `0`, and `set_config power` then writes the whole message back to the radio, overwriting whatever the node had.

That prompted an audit of all 23 `protoConfig` extensions against the fields declared in the pinned protobufs submodule. `sds_secs` is not isolated.

### Unmapped active fields

| Message | Unmapped |
| --- | --- |
| `Config.PowerConfig` | `sdsSecs`, `powermonEnables` |
| `Config.LoRaConfig` | `femLnaMode`, `ignoreIncoming`, `paFanDisabled`, `serialHalOnly` |
| `ModuleConfig.TelemetryConfig` | `airQualityScreenEnabled`, `healthMeasurementEnabled`, `healthScreenEnabled`, `healthUpdateInterval` |
| `Config.DisplayConfig` | `enableMessageBubbles`, `useLongNodeName`, `wakeOnTapOrMotion` |
| `Config.DeviceConfig` | `buzzerMode` |
| `Config.NetworkConfig` | `ipv6Enabled` |
| `ModuleConfig.RangeTestConfig` | `clearOnReboot` |

Each of these is written as a proto default on export and pushed back to the radio on import.

`LoRaConfig` is the one I would look at first: `femLnaMode` and `paFanDisabled` describe RF front-end hardware, so getting them wrong has physical consequences rather than cosmetic ones.

A further 4 flagged fields are **deprecated** and correctly skipped — `DeviceConfig.serial_enabled` (moved to `SecurityConfig`, which the exporter does write), `DisplayConfig.gps_format`, `PositionConfig.gps_attempt_time`, `CannedMessageConfig.allow_input_source`.

### Why the existing tests do not catch this

`DeviceProfileImportTests` verifies coverage parity at **section** level — all 8 device and 15 module configs are present. The round-trip test serializes and re-parses, which cannot notice a field that was never written in the first place, because it is absent from both sides of the comparison.

Every hardware import I ran reported success, and every firmware log showed the expected `Set config` / `Delay save` / single commit. Nothing surfaced this except a config diff against a snapshot taken before the import.

### Two ways to fix it

**Model the fields.** Add each to its SwiftData entity, ingest in `UpdateSwiftData.swift`, assign in `DeviceProfileExport.swift`. Correct and explicit, but it is 16 new attributes and a schema migration, and it leaves the same trap open for the next field added upstream.

**Merge onto the live config on import.** This PR already has the pattern, at `DeviceProfileImport.swift:262`:

```swift
static func securityConfig(from profile: Config.SecurityConfig, base: Config.SecurityConfig?)
```

which preserves the node's keypair when the profile does not carry one. Generalising that — merging each incoming config onto the node's current values — makes an unmodeled field keep whatever the radio already had, fixing all 16 at once and future ones for free.

The catch is real: proto3 scalars have no presence, so "the profile omits this" and "the profile sets this to 0" are indistinguishable. The security merge sidesteps it because keys are `bytes` and it can test `isEmpty`. For `sdsSecs`, treating 0 as "keep current" is almost certainly right. For every scalar in every config, it needs thought — and it would mean an import can no longer set any field to zero.

A third option worth naming: do neither now, and document the gap. The feature is still a large improvement over having no import at all, and these fields were equally absent from the export before this PR.

I do not think this should be decided quietly, hence raising it here rather than pushing a fix. My inclination is the merge, scoped initially to the fields in the table rather than applied blanket, but I would rather have a second opinion on the presence trade-off.

### Method, and its limits

The audit compares each `var protoConfig` block against the generated `.pb.swift` decode switch, which names every field exactly once per message regardless of whether SwiftProtobuf uses heap `_storage`. Read-only; the submodule and generated sources were not touched.

It is a regex over generated Swift, and it was wrong twice before it was right:

- v1 read `public var` declarations, so every `_storage`-backed message was invisible — `LoRaConfig` disappeared entirely — and it attributed nested message fields to their parent, reporting `ip`/`gateway`/`subnet`/`dns` as missing from `NetworkConfig` when they live on `NetworkConfig.IpV4Config`.
- Its self-check then "failed" on `DeviceConfig.serialEnabled`, which turned out to be the audit being right and my expectation being wrong: the field exists on both messages, and the exporter writes the `SecurityConfig` one.

All 16 in the table have since been individually confirmed with a direct grep for `config.<field>` in `DeviceProfileExport.swift`, and deprecation status comes from the `.proto` sources. Still, a second pair of eyes on the list before it drives a change would be sensible.

cc: @garthvh

**Comment by bruschill** [comment]
Two updates from re-testing against firmware develop @ `597f6767b` (135 commits past the `0199a1f` build used in the table above):

**1. The 2.8.0 "lossy link" caveat is resolved upstream.** meshtastic/firmware#11245 is fixed by meshtastic/firmware#11253 — confirmed on the same physical Heltec V4 with the NodeDB untouched: 693 dropped notifications over 392 s on `0199a1f`, zero over a full iOS sync on `597f676` (details in [the issue](https://github.com/meshtastic/firmware/issues/11245#issuecomment-5136715066)). The nRF52 `fromRadioQ full` / `tophone queue is full` saturation noted in the platform-difference section is also gone on develop. Transaction semantics were re-verified unchanged on `597f676` on both ESP32-S3 and nRF52840 — begin ×2 accepted, N deferred saves, exactly one flash save, one reboot — so scenario 6a's conclusion holds on current develop, now over a healthy link.

**2. Edge case observed: a security write inside a transaction can drop later writes.** During test cleanup on `597f676` (ESP32-S3, serial API), a single transaction batching 5 config writes ending with `security.debug_log_api_enabled = false` applied only 2 of them — the security write triggered `Disable bluetooth until reboot` and a PhoneAPI close mid-transaction, and the remaining writes never landed. A follow-up transaction applied them fine. I have not reduced this to a minimal repro yet, so treat it as observed-once. Relevance here: the import plan already orders Security late with only Channels & LoRa after it, which limits the blast radius, but it reinforces why the result screen reports items as *sent* rather than applied and why post-import verification exists. Worth a targeted repro before anyone leans harder on batching security with other sections.

**Comment by bruschill** [comment]
Follow-up for the in-flight cancel gap discussed above: bruschill/Meshtastic-Apple#22 makes transport sends cancellation-aware (keyed BLE continuations, TCP send cancel) and shields the transaction commit from inherited cancellation, cutting the force-dismiss backstop from 90s to 15s. Stacked on this branch; no changes to this PR's diff.

File: docs/user/settings.md [hunk 1]
```diff
@@ -135,6 +135,43 @@ The Traffic Management module helps reduce unnecessary mesh traffic and improve
...
+| Network | Wi-Fi / Ethernet settings |
+| Modules | All module settings (MQTT, Telemetry, Canned Messages, and so on) |
+| Ringtone & Canned Messages | Notification melody and canned-message text |
+| Fixed Position | A saved fixed GPS position |
+| Security & Identity | Private key, admin keys, and admin access |
+| Channels & LoRa | Channels and LoRa/region settings |
+
+All present sections are selected by default **except Security & Identity**, which is off by default: importing it replaces this node's cryptographic identity (its private/public key) and admin keys, which can break existing direct messages and lock you out of local administration. Leave it off when cloning a configuration onto a second radio that should keep its own identity.
+
+> **Warning — Import overwrites settings on your radio**
```
[the remark this answers is not in the document]
Comment (garthvh) [reply docs/user/settings.md]: Shortened to "Warning — Import replaces settings" (3 words, matching the sibling callout style) and regenerated the bundled docs.

**Comment by spitfire** [comment]
SI there a TestFlight version of the app so we can test it?

**Comment by bruschill** [comment]
> SI there a TestFlight version of the app so we can test it?

Not yet, but I'm sure there will be one eventually in the next few days. 👍🏻
